lib/rspec/mocks/error_generator.rb

```diff
@@ -291,6 +303,19 @@ def format_received_args(*args)
         args.empty? ? "(no args)" : "(" + received_arg_list(*args) + ")"
       end
 
+      def format_unexpected_with_args(*args)
+        with_args = Hash[Array(args.first).group_by { |x| x }.map { |k, v| [k, v.count] }]
+        received_with_arg_list(*with_args)
+      end
+
+      def received_with_arg_list(*args)
+        list = args.map do |a, i|
+          times_count = args.size > 1 ? '(' + times(i) + ')' : ''
+          ["(#{received_arg_list(*a)})", "#{times_count}"].join(" ")
```

Comment:
Also, I think you want to use `format_received_args` rather than `received_arg_list`, as the former takes care of putting `"no args"` in the string when there are none.  Having a spec that covers this case would be nice, although if you do refactor things above to have only one `raise_unexpected_message...` method (as I suggested above), then I suspect that the existing specs that cover that method will cover this case and will only pass if you switch to `format_received_args` -- in which case there's no need for an explicit spec.

Given that it's easy to use `received_arg_list` when `format_received_args` should be used, I'm thinking that maybe we should roll the `received_arg_list` logic in to `format_received_args` so that it's not even available to be called.

Thoughts?